src/Commands/Connection/SetCommand.php

```diff
@@ -73,7 +73,18 @@ public function connectionSet($site_env, $mode)
             throw $e;
         }
 
-        $this->processWorkflow($workflow);
+        try {
+            $this->processWorkflow($workflow);
+        } catch (TerminusException $e) {
+            if (strpos($e->getMessage(), 'build_status is building') !== false) {
+                throw new TerminusException(
+                    'Failed setting connection mode due to the environment being either in a broken or ' .
+                    'building state. ' .
```

Comment:
```suggestion
                    'Cannot switch to SFTP mode because the most recent Integrated Composer build ' .
                    'is either still running or encountered errrors.' .
```